### ISSUE TITLE
Choose Tech Stack

### ISSUE THREAD

**Issue by elliottsencan** [giving-a-fuck-about-climate-change/carbon-doomsday#5]
Any preferences on how we build this?

https://github.com/reactjs/react-chartjs jumps to mind, but i'm all ears

**Comment by lwm** [comment]
Since I have no idea what a front-end is and you're the only one so far, it's up to you! :)

**Comment by titojankowski** [comment]
What does coinbase/charts use?

**Comment by titojankowski** [comment]
Your choice, @elliottsencan!

fwiw Coinbase uses nvd3 for their chart: https://github.com/novus/nvd3

**Comment by lwm** [comment]
@grady-lad, you have some input here? I think you and @elliottsencan can wrangle this one out.

**Comment by lwm** [comment]
Actually, I better weigh in here as well. I'd use Python+Django to write the back-end (scripts to scrape data and expose the web API). Objections?

**Comment by grady-lad** [comment]
I would suggest just using d3.js, it has a bit of a learning curve initially but because it's such a low level canvas library you have the flexibility to build anything @elliottsencan what do you think ?

**Comment by titojankowski** [comment]
Great, want to set up a prototype using the 1975-2015 MOL dataset? @grady-lad

**Comment by elliottsencan** [comment]
@grady-lad thats another good option too. @lwm thats what i would do :D

I think it depends on where we want to go. A quick MVP would benefit from a higher level chart abstraction, the coinebase.com/chart example is pretty simple

Product Q: What kind of other charts do we want down the line?

1. Carbon footprint per person
2. Ice cap area over time
3. sea level visualization
4. Trends based on international agreements

d3 is a better choice if we start getting into more complex visualizations

**Comment by titojankowski** [comment]
You're right that there's potential in tracking sea ice, sea levels, and other metrics. One challenge is those climate change metrics are not currently as quantified or standardized as CO2. Carbon dioxide is the gold standard, measured hourly on top of Mauna Loa since 1975.

One outcome of our first MVP may be we connect with leaders in these topics who can help build out more complex offerings. Another is to double down on CO2 and connect to different sampling sites around the world, though the data will probably be similar it could help people understand local CO2 events.

The way to unlock either of those possibilities lies in the big juicy problem that the existing graphs for carbon dioxide (the gold standard!) are inaccessible and uninteresting. Our chart will change all that and unlock further opportunities.

Let's get the coinbase-inspired CO2 MVP prototyped and launched, get eyeballs, collaborators, and media and keep our eyes peeled on what emerges from there.

**Comment by titojankowski** [comment]
Attached is a mockup for the MVP. Does this help answer your questions, or perhpas spark new ones?

![img_5184](https://user-images.githubusercontent.com/129060/27006085-b97b253e-4de0-11e7-84a8-798b66e30a87.JPG)

**Comment by lwm** [comment]
> Let's get the coinbase-inspired CO2 MVP prototyped and launched, get eyeballs, collaborators, and media and keep our eyes peeled on what emerges from there.

Yep, marked this on https://github.com/giving-a-fuck-about-climate-change/carbondoomsday-django#current-misson :)

**Comment by lwm** [comment]
Looks like we can bundle real-time comms (django-channels), react and d3 all together to get a really speedy app in the one place. I've seen django, django-channels and react all put in place [in this tutorial](https://github.com/codyparker/channels-obstruction). 

Then, moving onto getting something like [the mockup](https://files.gitter.im/giving-a-fuck-about-climate-change/Lobby/RC8C/Sketch1.png) we can do [d3 live updates](http://bl.ocks.org/d3noob/7030f35b72de721622b8) for week/month/year buttons. The rest is just loading current PPM and increase for last week, month, year etc.
